types/mathjs/index.d.ts

```diff
@@ -1,2252 +1,4565 @@
+         * @returns Returns the created index
+         */
+        index(...ranges: any[]): Index;
+
+        /**
+         * Create a Matrix. The function creates a new math.type.Matrix object
+         * from an Array. A Matrix has utility functions to manipulate the data
+         * in the matrix, like getting the size and getting or setting values in
+         * the matrix. Supported storage formats are 'dense' and 'sparse'.
+         * @param format The Matrix storage format
+         * @returns The created Matrix
+         */
+        matrix(format?: "sparse" | "dense"): Matrix;
+        /**
+         * @param data A multi dimensional array
+         * @param format The Matrix storage format
+         * @param dataType The Matrix data type
+         * @returns The created Matrix
+         */
+        matrix(
+            data: MathArray | Matrix,
+            format?: "sparse" | "dense",
+            dataType?: string
+        ): Matrix;
+
+        /**
+         * Create a number or convert a string, boolean, or unit to a number.
+         * When value is a matrix, all elements will be converted to number.
+         * @param value Value to be converted
+         * @returns The created number
+         */
+        number(
+            value?:
+                | string
+                | number
+                | BigNumber
+                | Fraction
+                | boolean
+                | MathArray
+                | Matrix
+                | Unit
+                | null
+        ): number | MathArray | Matrix;
+        /**
+         * @param value Value to be converted
+         * @param valuelessUnit A valueless unit, used to convert a unit to a
+         * number
+         * @returns The created number
+         */
+        number(unit: Unit, valuelessUnit: Unit | string): number;
+
+        /**
+         * Create a Sparse Matrix. The function creates a new math.type.Matrix
+         * object from an Array. A Matrix has utility functions to manipulate
+         * the data in the matrix, like getting the size and getting or setting
+         * values in the matrix.
+         * @param data A two dimensional array
+         * @param dataType Sparse Matrix data type
+         * @returns The created matrix
+         */
+        sparse(data?: MathArray | Matrix, dataType?: string): Matrix;
+
+        /**
+         * Split a unit in an array of units whose sum is equal to the original
+         * unit.
+         * @param unit A unit to be split
+         * @param parts An array of strings or valueless units
+         * @returns An array of units
+         */
+        splitUnit(unit: Unit, parts: Unit[]): Unit[];
+
+        /**
+         * Create a string or convert any object into a string. Elements of
+         * Arrays and Matrices are processed element wise.
+         * @param value A value to convert to a string
+         * @returns The created string
+         */
+        string(
+            value: any | MathArray | Matrix | null
```

Comment:
Should the return type of this should just be `string`?

Reply:
http://mathjs.org/docs/reference/functions/string.html Mathjs docs show that it can return MathArray or Matrix... don't exactly know why